ios/RCTWebRTC/WebRTCModule+Daily.m

```diff
@@ -144,58 +145,22 @@ - (void)audioSession:(RTCAudioSession *)audioSession didSetActive:(BOOL)active {
   // Apply the chosen audio mode right away if the audio session is already
   // active. Otherwise, it will be applied when the session becomes active.
   RTCAudioSession *audioSession = RTCAudioSession.sharedInstance;
+  NSLog(@"[Daily] setDailyAudioMode: %@", audioMode);
   if (audioSession.isActive) {
     [self applyAudioMode:audioMode toSession:audioSession];
   }
 }
 
 - (void)applyAudioMode:(NSString *)audioMode toSession:(RTCAudioSession *)audioSession {
+  NSLog(@"[Daily] applyAudioMode: %@", audioMode);
   // Do nothing if we're attempting to "unset" the in-call audio mode (for now
   // it doesn't seem like there's anything to do).
   if ([audioMode isEqualToString:AUDIO_MODE_IDLE]) {
     return;
   }
   
-  // Ducking other apps' audio implicitly enables allowing mixing audio with
-  // other apps, which allows this app to stay alive in the backgrounnd during
-  // a call (assuming it has the voip background mode set).
-  AVAudioSessionCategoryOptions categoryOptions = (AVAudioSessionCategoryOptionAllowBluetooth |
-                                                   AVAudioSessionCategoryOptionDuckOthers);
-  if ([audioMode isEqualToString:AUDIO_MODE_VIDEO_CALL]) {
-    categoryOptions |= AVAudioSessionCategoryOptionDefaultToSpeaker;
-  }
-  [self audioSessionSetCategory:AVAudioSessionCategoryPlayAndRecord toSession:audioSession options:categoryOptions];
-  
-  
-  NSString *mode = ([audioMode isEqualToString:AUDIO_MODE_VIDEO_CALL] ?
-                    AVAudioSessionModeVideoChat :
-                    AVAudioSessionModeVoiceChat);
-  [self audioSessionSetMode:mode toSession:audioSession];
-}
-
-- (void)audioSessionSetCategory:(NSString *)audioCategory
-                      toSession:(RTCAudioSession *)audioSession
-                        options:(AVAudioSessionCategoryOptions)options
-{
-  @try {
-    [audioSession setCategory:audioCategory
-                  withOptions:options
-                        error:nil];
-    NSLog(@"[Daily] audioSession.setCategory: %@, withOptions: %lu success", audioCategory, (unsigned long)options);
-  } @catch (NSException *e) {
-    NSLog(@"[Daily] audioSession.setCategory: %@, withOptions: %lu fail: %@", audioCategory, (unsigned long)options, e.reason);
-  }
-}
-
-- (void)audioSessionSetMode:(NSString *)audioMode
-                  toSession:(RTCAudioSession *)audioSession
-{
-  @try {
-    [audioSession setMode:audioMode error:nil];
-    NSLog(@"[Daily] audioSession.setMode(%@) success", audioMode);
-  } @catch (NSException *e) {
-    NSLog(@"[Daily] audioSession.setMode(%@) fail: %@", audioMode, e.reason);
-  }
+  NSString * audioDevice = self.hasBluetoothDevice ? BLUETOOTH_DEVICE_ID : ([audioMode isEqualToString:AUDIO_MODE_VIDEO_CALL] ? SPEAKER_DEVICE_ID: WIRED_OR_EARPIECE_DEVICE_ID);
```

Comment:
Nit: format this line so it doesn't end up so long/hard to read

Reply:
What about `AUDIO_MODE_USER_SPECIFIED`? (Or whatever we decided to call it)

Reply:
Looks like `AUDIO_MODE_USER_SPECIFIED` isn't anywhere in this file...

Reply:
Different from Android, where we were using a listener to identify each time a device was added or removed and invoking to reconfigure, we hadn't previously any kind of special treatment for that on IOs. 

We were just relying on whatever the iOS does by default.

So, in this case, I didn't need to add any special treatment here, so we don't change the iOS default behavior.

Reply:
Fixed. Take a look on my last commit: _Refactor to keep the user selected device, and don't change the audio mode unless the user has explicitly changed the device._